apps/music/elements/music-search-box.js

```diff
@@ -143,54 +90,66 @@ proto.attributeChangedCallback = function(attr, oldVal, newVal) {
 };
 
 proto.clear = function() {
-  window.requestAnimationFrame(() => {
-    this.els.form.reset();
-    this.els.input.focus();
+  requestAnimationFrame(() => {
+    this.els.input.clear();
   });
 };
 
-proto.open = function() {
-  window.requestAnimationFrame(() => {
+proto.focus = function() {
+  requestAnimationFrame(() => {
     this.els.input.focus();
   });
+};
 
-  this.dispatchEvent(new CustomEvent('open'));
+proto.open = function() {
+  if (this.isOpen) return;
+  this.isOpen = true;
+  this.els.inner.classList.add('open');
+  this.focus();
+  this.emit('open');
 
   if (this.els.results) {
     this.els.results.open();
   }
+
 };
 
 proto.close = function() {
-  this.els.form.reset();
-
-  this.dispatchEvent(new CustomEvent('close'));
+  if (!this.isOpen) return;
+  this.isOpen = false;
+  this.els.inner.classList.remove('open');
+  this.clear();
+  this.emit('close');
 
   if (this.els.results) {
     this.els.results.close();
   }
+
 };
 
-Object.defineProperty(proto, 'value', {
-  get: function() {
-    return this.els.input.value;
-  },
+proto.emit = function(name, detail) {
```

Comment:
This goes against the pattern in our other custom elements where we just use `dispatchEvent` directly.